### PR TITLE
Add support for SPV_INTEL_joint_matrix extension

### PR DESCRIPTION
**Pull request by AlexeySotkin**

### CODE REVIEW

File: llvm-spirv/lib/SPIRV/libSPIRV/SPIRVModule.cpp [hunk 1]
```diff
@@ -897,6 +900,15 @@ SPIRVTypeVector *SPIRVModuleImpl::addVectorType(SPIRVType *CompType,
                                                 SPIRVWord CompCount) {
   return addType(new SPIRVTypeVector(this, getId(), CompType, CompCount));
 }
+
+SPIRVTypeJointMatrixINTEL *
+SPIRVModuleImpl::addMatrixINTELType(SPIRVType *CompType, SPIRVValue *Scope,
```
Comment (AlexeySotkin) [inline llvm-spirv/lib/SPIRV/libSPIRV/SPIRVModule.cpp:905]: Order of parameters seems to be wrong.

**Comment by yubingex007-a11y** [comment]
Hi, Will you create another PR for clang driver and adding spirv-ext for SPV_INTEL_joint_matrix ?

**Comment by AlexeySotkin** [comment]
> Hi, Will you create another PR for clang driver and adding spirv-ext for SPV_INTEL_joint_matrix ?

Yes

**Comment by dkhaldi** [comment]
LGTM

**Review by dkhaldi — APPROVED** [review]
LGTM

**Review by mdtoguchi — APPROVED** [review]
Driver LGTM

**Review by MrSidims — APPROVED** [review]

**Comment by AlexeySotkin** [comment]
Partially committed to the translator upstream repo.

**Pull request closed without merging**